mmeval/metrics/cityscapes_detection.py

```diff
@@ -0,0 +1,327 @@
+        >>> cityscapes_det_metric = CityScapesDetection(
+        ...     dataset_meta=dataset_metas,
+        ...     outfile_prefix=osp.join(tmp_dir.name, 'test'),
+        ...     seg_prefix=seg_prefix,
+        ...     keep_results=False,
+        ...     keep_gt_json=False,
+        ...     classwise=True)
+        >>>
+        >>> def _gen_fake_datasamples(seg_prefix):
+        ...     city = 'lindau'
+        ...     os.makedirs(osp.join(seg_prefix, city), exist_ok=True)
+        ...
+        ...     sequenceNb = '000000'
+        ...     frameNb1 = '000019'
+        ...     img_name1 = f'{city}_{sequenceNb}_{frameNb1}_gtFine_instanceIds.png'
+        ...     img_path1 = osp.join(seg_prefix, city, img_name1)
+        ...     basename1 = osp.splitext(osp.basename(img_path1))[0]
+        ...     masks1 = np.zeros((20, 20), dtype=np.int32)
+        ...     masks1[:10, :10] = 24 * 1000
+        ...     Image.fromarray(masks1).save(img_path1)
+        ...
+        ...     dummy_mask1 = np.zeros((1, 20, 20), dtype=np.uint8)
+        ...     dummy_mask1[:, :10, :10] = 1
+        ...     prediction1 = {
+        ...         'basename': basename1,
+        ...         'mask_scores': np.array([1.0]),
+        ...         'labels': np.array([0]),
+        ...         'masks': dummy_mask1
+        ...     }
+        ...     groundtruth1 = {
+        ...         'file_name': img_path1
+        ...     }
+        ...
+        ...     frameNb2 = '000020'
+        ...     img_name2 = f'{city}_{sequenceNb}_{frameNb2}_gtFine_instanceIds.png'
+        ...     img_path2 = osp.join(seg_prefix, city, img_name2)
+        ...     basename2 = osp.splitext(osp.basename(img_path2))[0]
+        ...     masks2 = np.zeros((20, 20), dtype=np.int32)
+        ...     masks2[:10, :10] = 24 * 1000 + 1
+        ...     Image.fromarray(masks2).save(img_path2)
+        ...
+        ...     dummy_mask2 = np.zeros((1, 20, 20), dtype=np.uint8)
+        ...     dummy_mask2[:, :10, :10] = 1
+        ...     prediction2 = {
+        ...         'basename': basename2,
+        ...         'mask_scores': np.array([0.98]),
+        ...         'labels': np.array([1]),
+        ...         'masks': dummy_mask1
+        ...     }
+        ...     groundtruth2 = {
+        ...         'file_name': img_path2
+        ...     }
+        ...     return [prediction1, prediction2], [groundtruth1, groundtruth2]
+        >>>
+        >>> predictions, groundtruths = _gen_fake_datasamples(seg_prefix)
+        >>> cityscapes_det_metric(predictions=predictions, groundtruths=groundtruths)  # doctest: +ELLIPSIS  # noqa: E501
+        {'mAP': ..., 'AP50': ...}
+        >>> tmp_dir.cleanup()
+    """
+
+    def __init__(self,
+                 outfile_prefix: str,
+                 seg_prefix: Optional[str] = None,
+                 format_only: bool = False,
+                 keep_results: bool = False,
+                 keep_gt_json: bool = False,
+                 classwise: bool = False,
```

Comment:
The default value should be considered. What's the convention on this dataset/metric?